frontend/src/streamDetails/__tests__/StreamStatisticsCard.test.jsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import { describe, expect, it } from "@jest/globals";
+
+import { Grid } from "@material-ui/core";
+import StreamStatisticsCard from "../StreamStatisticsCard";
+import DashboardCard from "../../general/dashboard/DashboardCard";
+import SimpleTable from "../../general/simpleTable/SimpleTable";
+import { getSampleStreamStats } from "../../api/SampleData";
+import ButtonInfo from "../../general/dashboard/ButtonInfo";
+
+Enzyme.configure({ adapter: new Adapter() });
+
+describe("<StreamStatisticsCard/> class component", () => {
+  let wrapper;
+  const dummyStreamId = 1;
+  const dummyStats = getSampleStreamStats();
+
+  const expectedProperties = {
+    Time: dummyStats.time,
+    "Round-Trip-Time": dummyStats.link.rtt,
+    "Packets Retransmitted": dummyStats.send.packetsRetransmitted,
+    "Packets Dropped": dummyStats.send.packetsDropped
+  };
+
+  const expectedButton = new ButtonInfo(
+    `/Streams/Details/${dummyStats.id}/Statistics`,
+    { statistics: dummyStats },
+    "More Statistics"
+  );
+
+  beforeEach(() => {
+    wrapper = Enzyme.shallow(<StreamStatisticsCard streamId={dummyStreamId} />);
+  });
+  afterEach(() => {
+    wrapper.unmount();
+  });
+  describe("handleStatsChange() function", () => {
+    it("should set the stats state", () => {
+      const startingState = {
+        stats: []
+      };
+      const expectedState = {
+        stats: dummyStats
+      };
+
+      expect(wrapper.state()).toStrictEqual(startingState);
```

Comment:
instead of asserting this, you can directly wrapper.setState(startingState) in order to set it before you change it
(Though I absolutely recognize that this is leftover from how I used to test things)

Reply:
i left it like this because i wanted to check that the default starting state should be `[]` without having to be explicitly set. setting the state myself means if there were an error with the default state not being set at all this test would miss it cuz we force the state assignment? 
idk, i don't know anything about testing. i can remove the check entirely or do what you mentioned if you want

Reply:
if you want to test the default state, it's more clear if you add another test for that. 

This test should be checking that handleStatsChange() sets state.stats value. And , since you're checking that the value changed, it's better to set it to a different value first (to verify that there was a change) as part of the Arrange=> Act=> Assert structure of tests

Reply:
e57e099